Wikipedia/Code/LoadingFlowController.swift

```diff
@@ -0,0 +1,368 @@
+                let mwkArticle = article as? MWKArticle {
+                articleVC.skipFetchOnViewDidAppear = true
+                wmf_add(childController: articleVC, andConstrainToEdgesOfContainerView: view)
+                articleVC.article = mwkArticle
+                articleVC.kickoffProgressView()
+            } else {
+                assertionFailure("Issue pushing article view controller")
+            }
+        case .linkPush:
+            
+            let articleVC = WMFArticleViewController(articleURL: url, dataStore: dataStore, theme: theme)
+            let loadingFlowController = LoadingFlowController(dataStore: dataStore, theme: theme, fetchDelegate: articleVC, flowChild: articleVC, url: url, embedOnLoad: true)
+            articleVC.loadingFlowController = loadingFlowController
+            
+            if let mwkArticle = article as? MWKArticle {
+                articleVC.viewDidLoadCompletion = {
+                    articleVC.article = mwkArticle
+                }
+                
+                articleVC.skipFetchOnViewDidAppear = true
+            }
+            
+            wmf_push(loadingFlowController, animated: true)
+        }
+    }
+    
+    private func showTalkPage(url: URL, source: ProcessSource) {
+        
+        guard let siteURL = url.wmf_site else {
+            assertionFailure("Issue determining siteURL for talk page.")
+            return
+        }
+        
+        var title = url.lastPathComponent
+        
+        if let firstColon = title.range(of: ":") {
+            title.removeSubrange(title.startIndex..<firstColon.upperBound)
+        }
+        
+        let titleWithTalkPageNamespace = TalkPageType.user.titleWithCanonicalNamespacePrefix(title: title, siteURL: siteURL)
+        
+        switch source {
+        case .loadEmbed:
+            assertionFailure("Talk page container not setup to embed")
+        case .linkPush:
+            let containerVC = TalkPageContainerViewController.containedTalkPageContainer(title: titleWithTalkPageNamespace, siteURL: siteURL, dataStore: dataStore, type: .user, theme: theme)
+            self.navigationController?.pushViewController(containerVC, animated: true)
+        }
+    }
+}
+
+extension LoadingFlowController: SFSafariViewControllerDelegate {
+    func safariViewControllerDidFinish(_ controller: SFSafariViewController) {
+        self.navigationController?.popViewController(animated: true)
+    }
+}
+
+extension LoadingFlowController: Themeable {
+    
+    func apply(theme: Theme) {
+        self.theme = theme
+        guard viewIfLoaded != nil else {
+            return
+        }
+        
+        view.backgroundColor = theme.colors.paperBackground
+        loadingAnimationViewController.apply(theme: theme)
+        flowChild.apply(theme: theme)
+        flowChild.customNavAnimationHandler?.apply(theme: theme)
+        wmf_applyTheme(toEmptyView: theme)
+    }
+}
+
+//MARK: Error Handling
+
+extension NSError {
```

Comment:
Was this meant to be an internal extension or could it be private?

Reply:
@nambatee Good point, I've made this private.